src/qengine/opencl.cpp

```diff
@@ -2024,6 +2024,17 @@ complex QEngineOCL::GetAmplitude(bitCapInt fullRegister)
     return amp[0];
 }
 
+void QEngineOCL::SetAmplitude(bitCapInt perm, complex amp)
+{
+    runningNorm -= norm(GetAmplitude(perm));
```

Comment:
Shouldn't this be `=`, instead of `-=`, since the amplitude is being locked here?

Reply:
Actually, no, the `runningNorm` is the total on-the-fly normalization constant, and it should be modified by the _differential_ from the single amplitude, as this intends, but there's still a slight problem that I'll fix.

Reply:
Did the slight problem get fixed?

Reply:
And, more importantly, have a unit test added to help catch similar bugs in the future?

Reply:
This would potentially be really hard to isolate, because of (good, computationally cheap) "redundancy" on iterative normalization when the option for normalization is turned out all. However, it's fixed, and it's hard to argue that our coverage is insufficient. (Look at what actually runs in the CI, which, by the way, could now have Q# tests added to it, as I work that out in the new repo on the other end of this feature.)